3dw1nM0535/maintenance-tracker-app
# Ch test cases 157930650

# Test cases for User authentication

* Create tests for registering a user without encountering an error
* Create tests for logging in a user without encountering an error
* Create tests for authenticating a user using his/her password
* Create tests for validating user email(should not be empty)
* Create tests for validating password input(should not be empty)
* Create tests for email duplication(should not have duplicate emails in our database)
* Create tests for validating input during signup
* Create tests for validating input during sign in

To test this PR
* `git clone ` to have a local copy of the project
* `cd maintenance-tracker-app` to navigate into the cloned project
* `pip install virtualenv` for creating isolated Python environments
* `virtualenv venv` to create a virtual environment to work in
* `. venv/bin/activate` to activate your virtual environment
* `pip install -r requirements.txt` to install project's packages
* `git checkout ch-test-cases-157930650` to checkout from develop
* `nosetests --rednose` to run the tests

Relevant pivotal tracker story
[#157930650](https://www.pivotaltracker.com/story/show/157930650)
